pkg/udp/conn.go

```diff
@@ -33,18 +34,18 @@ type Listener struct {
 }
 
 // Listen creates a new listener.
-func Listen(network string, laddr *net.UDPAddr, timeout time.Duration) (*Listener, error) {
+func Listen(listenConfig net.ListenConfig, network, address string, timeout time.Duration) (*Listener, error) {
 	if timeout <= 0 {
 		return nil, errors.New("timeout should be greater than zero")
 	}
 
-	conn, err := net.ListenUDP(network, laddr)
+	packetConn, err := listenConfig.ListenPacket(context.Background(), network, address)
 	if err != nil {
 		return nil, err
 	}
 
 	l := &Listener{
-		pConn:     conn,
+		pConn:     packetConn.(*net.UDPConn),
```

Comment:
Actually, `packetConn` is always a `*net.UDPConn` in this case, so there's no risk of panic. But I've made the change you requested, since adding the nil check could provide extra reassurance.